### PR TITLE
Add TCP based intents discovery

### PR DESCRIPTION
**Pull request by NetanelBollag**
### Description

Add TCP-SYN packet sniffing to allow discovery of intents from pure TCP communication.

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### CODE REVIEW

File: src/shared/config/config.go [hunk 1]
```diff
@@ -19,6 +19,12 @@ const (
 	PrometheusMetricsPortDefault = 2112
 	TelemetryErrorsAPIKeyKey     = "telemetry-errors-api-key"
 	TelemetryErrorsAPIKeyDefault = "d86195588a41fa03aa6711993bb1c765"
+	EnableTCPKey                 = "enable-tcp"
+	EnableTCPSnifferDefault      = true
+	EnableSocketScannerKey       = "enable-socket-scanner"
+	EnableSocketScannerDefault   = true
+	EnableDNSKey                 = "enable-dns"
+	EnableDNSSnifferDefault      = true
```
Comment (evyatarmeged) [inline src/shared/config/config.go:27]: So we're having flags for each type now ?

File: src/sniffer/pkg/collectors/tcpsniffer.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+func (s *TCPSniffer) HandlePacket(packet gopacket.Packet) {
+	if !viper.GetBool(sharedconfig.EnableTCPKey) {
+		return
+	}
+	captureTime := detectCaptureTime(packet)
+	ipLayer := packet.Layer(layers.LayerTypeIPv4)
+	if ipLayer != nil {
+		ip, _ := ipLayer.(*layers.IPv4)
```
Comment (evyatarmeged) [inline src/sniffer/pkg/collectors/tcpsniffer.go:77]: Check for ok here or is it always ipv4 ?

File: src/sniffer/pkg/collectors/tcpsniffer.go [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+	for _, p := range s.pending {
+		hostname, err := s.resolver.ResolveIP(p.srcIp)
+		if err != nil {
+			logrus.WithError(err).Debugf("Could not to resolve %s, skipping packet", p.srcIp)
+			continue
+		}
+		if p.srcHostname != hostname {
+			logrus.Debugf("IP %s was resolved to %s, but now resolves to %s. skipping packet", p.srcIp, p.srcHostname, hostname)
+			continue
```
Comment (evyatarmeged) [inline src/sniffer/pkg/collectors/tcpsniffer.go:108]: Chef's kiss :ok_hand:

File: src/shared/config/config.go (hunk 1 above)
Comment (NetanelBollag) [reply src/shared/config/config.go:27]: Yes, it's can be helpful for debugging, shouldn't be used for anything else

File: src/sniffer/pkg/collectors/tcpsniffer.go (hunk 2 above)
Comment (NetanelBollag) [reply src/sniffer/pkg/collectors/tcpsniffer.go:77]: I guess returning && printing debug log won't hurt

**Pull request merged**